libs/components/molecules/src/lib/interactive-svg/interactive-svg.component.ts

```diff
@@ -106,6 +116,53 @@ export class InteractiveSvgComponent<T extends NodeMapEntry> implements OnDestro
   /** Destroys */
   private destroy$ = new Subject<void>();
 
+  /** Crosswalk element of svg */
+  private crosswalkEl?: Element;
+
+  /**
+   * Updates the highlighting based on current highlight id
+   * @param changes
+   */
+  ngOnChanges(changes: SimpleChanges): void {
+    if ('highlightId' in changes) {
+      this.resetHighlight();
+      this.setHighlight();
+    }
+  }
+
+  /**
+   * Highlights cells that match highlightId
+   */
+  private setHighlight() {
```

Comment:
This method needs some refactoring/cleanup to reduce nesting/duplication. @edlu77 Ping me on slack and we can meet to go over it